src/hooks/useAutoCompound.ts

```diff
@@ -0,0 +1,225 @@
+    const api = $api;
+    const batchTxsRef = batchTxs;
+    const oakAPI = await getAPI(oakEndpoint.value);
+
+    const txResHandler = async (result: ISubmittableResult): Promise<boolean> => {
+      const res = await handleResult(result);
+      hasExtrinsicFailedEvent(result.events, store.dispatch);
+      return res;
+    };
+
+    if (!api) {
+      throw Error('Failed to connect to API');
+    }
+    if (0 >= batchTxsRef.length) {
+      throw Error('No dApps can be claimed');
+    }
+
+    const txsToExecute: ExtrinsicPayload[] = [];
+    let totalWeight: BN = new BN(0);
+    for (let i = 0; i < batchTxsRef.length; i++) {
+      const tx = batchTxsRef[i];
+      if (totalWeight.add(tx.weight).gt(MAX_BATCH_WEIGHT)) {
+        break;
+      }
+
+      txsToExecute.push(tx.payload as ExtrinsicPayload);
+      totalWeight = totalWeight.add(tx.weight);
+    }
+
+    txsToExecute.unshift(api.tx.dappsStaking.setRewardDestination(RewardDestination.FreeBalance));
+    txsToExecute.push(api.tx.dappsStaking.setRewardDestination(RewardDestination.StakeBalance));
+
+    console.info(
+      `Batch weight: ${totalWeight.toString()}, transactions no. ${txsToExecute.length}`
+    );
+    const proxyExtrinsic = api.tx.utility.batch(txsToExecute);
+
+    const injector = await getInjector(substrateAccounts.value);
+    if (!injector) {
+      throw Error('Invalid injector');
+    }
+
+    const address = senderAddress.value;
+    const proxyCall = api.tx.proxy.proxy(address, 'Any', proxyExtrinsic);
+    const encodedProxyCall = proxyCall.method.toHex();
+    const proxyCallFees = await proxyCall.paymentInfo(address);
+
+    // If there is no proxy, add proxy.
+    const proxiesResponse = await api.query.proxy.proxies(address);
+
+    if (isEmpty(proxiesResponse)) {
+      const proxyAddress = getProxyAccount(api, address, oakParaId.value);
+      const proxyTxn = api.tx.proxy.addProxy(proxyAddress, 'Any', 0);
+
+      // TODO: add tip, show error
+      await proxyTxn.signAndSend(
+        address,
+        {
+          signer: injector.signer,
+          nonce: -1,
+        },
+        (result) => {
+          (async () => {
+            const res = await txResHandler(result);
+          })();
+        }
+      );
+    }
+
+    // Schedule XCM call
+    const providedId = `xcmp_astar_automation_${(Math.random() + 1).toString(36).substring(7)}`;
+    const xcmpCall = oakAPI.tx.automationTime.scheduleXcmpTask(
+      providedId,
+      [Math.floor(Date.now() / 1000) + 24 * 60 * 60], // TODO: calculate execution time
+      selfParaId.value,
+      0,
+      encodedProxyCall,
+      proxyCallFees.weight
+    );
+
+    console.log(xcmpCall);
+
+    // Get task id
+    // const turingAddress = encodeAddress(address, oakSS58Prefix.value);
+    // const taskId = await oakAPI.rpc.automationTime.generateTaskId(turingAddress, providedId);
+
+    // Sign and send scheduleXcmpTask call
+    await xcmpCall.signAndSend(
```

Comment:
Let's define the new service in [v2](https://github.com/AstarNetwork/astar-apps/tree/main/src/v2) folder and call from service.

example (stake transaction):

https://github.com/AstarNetwork/astar-apps/blob/main/src/hooks/dapps-staking/useStake.ts#L62

https://github.com/AstarNetwork/astar-apps/blob/main/src/v2/services/IDappStakingService.ts#L22